sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -388,99 +390,127 @@ impl Plan {
-                let zone = dns_builder.host_zone(id, address).unwrap();
+                let zone = dns_builder.host_zone(id, ip).unwrap();
                 dns_builder
                     .service_backend_zone(ServiceName::Cockroach, &zone, port)
                     .unwrap();
-                let address = SocketAddrV6::new(address, port, 0, 0);
-                request.datasets.push(DatasetEnsureBody {
+                request.services.push(ServiceZoneRequest {
                     id,
-                    zpool_id: u2_zpools[0],
-                    dataset_kind: crate::params::DatasetKind::CockroachDb,
-                    address,
+                    zone_type: ZoneType::CockroachDb,
+                    addresses: vec![ip],
+                    dataset: Some(DatasetRequest {
+                        id,
+                        name: DatasetName::new(
+                            u2_zpools[0].clone(),
+                            crate::params::DatasetKind::CockroachDb,
+                        ),
+                    }),
+                    gz_addresses: vec![],
+                    services: vec![ServiceZoneService {
+                        id,
+                        details: ServiceType::CockroachDb,
+                    }],
                 });
             }
 
             // TODO(https://github.com/oxidecomputer/omicron/issues/732): Remove
             if idx < CLICKHOUSE_COUNT {
                 let id = Uuid::new_v4();
-                let address = addr_alloc.next().expect("Not enough addrs");
+                let ip = addr_alloc.next().expect("Not enough addrs");
                 let port = omicron_common::address::CLICKHOUSE_PORT;
-                let zone = dns_builder.host_zone(id, address).unwrap();
+                let zone = dns_builder.host_zone(id, ip).unwrap();
                 dns_builder
                     .service_backend_zone(ServiceName::Clickhouse, &zone, port)
                     .unwrap();
-                let address = SocketAddrV6::new(address, port, 0, 0);
-                request.datasets.push(DatasetEnsureBody {
+                request.services.push(ServiceZoneRequest {
                     id,
-                    zpool_id: u2_zpools[0],
-                    dataset_kind: crate::params::DatasetKind::Clickhouse,
-                    address,
+                    zone_type: ZoneType::Clickhouse,
+                    addresses: vec![ip],
+                    dataset: Some(DatasetRequest {
+                        id,
+                        name: DatasetName::new(
+                            u2_zpools[0].clone(),
+                            crate::params::DatasetKind::Clickhouse,
+                        ),
+                    }),
+                    gz_addresses: vec![],
+                    services: vec![ServiceZoneService {
+                        id,
+                        details: ServiceType::Clickhouse,
+                    }],
```

Comment:
When scaling up ClickHouse, the coordinator nodes (keepers) will need to be installed in separate hosts than the databases (main and replica). We'll need to have a separate service for this from what I can tell.

Reply:
That tracks - if we want this service to be provisioned as a part of rack setup, this is where we could make that allocation.